### PR TITLE
Refactoring routes into sub functions

### PR DESCRIPTION
**Pull request by KevinDyer**
This is a larger scale change.

These changes also add mocha, chai, and supertest for testing the routes. I also moved the previous test harness into its own test file. Currently I'm `skip`ing those tests because I the tests do not match (I'm using the dev database locally and suspect there is some difference between dev and prod).

### CODE REVIEW

File: test/compare-compile-test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  testData.forEach((data) => {
+    // TODO(kevindy) These tests should be more hermetic (compile same data?)
+    it.skip(`should compile data: ${data}`, function(done) {
+      getLocalAndRemoteCompile(data, function(err, local, remote) {
+        if (err) {
+          return done(err);
+        }
+        done();
```
Comment (jeffdyer) [inline test/compare-compile-test.js]: Need to use jsonDiff here to compare local and remote response bodies.

File: test/compare-compile-test.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
+const LOCAL_GATEWAY = 'http://localhost:3000/';
+const REMOTE_GATEWAY = 'https://www.graffiticode.com/';
+
+const path = require('path');
+const url = require('url');
+const request = require('request');
+
+const testData = require(path.resolve(__dirname, './../tools/test.json'));
```
Comment (jeffdyer) [inline test/compare-compile-test.js]: FYI. I wrote the code to get the test cases from the currently connected database. I'll merge that code here after this pull request is merged.

Comment (KevinDyer) [reply test/compare-compile-test.js]: got it i thought i would get to it but definitely didnt have time today

File: test/compare-compile-test.js (hunk 1 above)
Comment (KevinDyer) [reply test/compare-compile-test.js]: Yep those were some unfinished test. New commit should add the check. These tests are being skipped at the moment. I think it makes sense to enable them once the test data is pulled from the db

**Pull request merged**